crates/biome_js_analyze/src/analyzers/nursery/use_shorthand_assign.rs

```diff
@@ -0,0 +1,268 @@
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let operator = node.operator().ok()?;
+        let right = node.right().ok();
+        let left = node.left().ok();
+        let left_var_name = left?.text();
+
+        if !matches!(operator, JsAssignmentOperator::Assign) {
+            return None;
+        }
+
+        let binary_expression = match right {
+            Some(AnyJsExpression::JsBinaryExpression(binary_expression)) => Some(binary_expression),
+            Some(AnyJsExpression::JsParenthesizedExpression(param)) => Some(
+                JsBinaryExpression::cast_ref(param.expression().ok()?.syntax())?,
+            ),
+            _ => None,
+        }?;
+
+        let has_same_reference =
+            has_same_reference_in_expression(left_var_name, &binary_expression)?;
+        let operator = binary_expression.operator().ok()?;
+        let right = binary_expression.right().ok()?;
+        let shorhand = get_shorthand(&operator)?;
+
+        if has_same_reference {
+            Some(RuleState {
+                shorthand_operator: shorhand,
+                right,
+            })
+        } else {
+            None
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        let shorthand_operator = state.shorthand_operator.to_string()?;
+
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {
+                "Assignment (=) can be replaced with operator assignment "{shorthand_operator}""
+            },
+        ))
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+        let mut mutation = ctx.root().begin();
+        let shorthand_operator = state.shorthand_operator;
+
+        let token_leading_trivia = node.operator_token().ok()?.leading_trivia().pieces();
+        let token_trailing_trivia = node.operator_token().ok()?.trailing_trivia().pieces();
+        let token = make::token(shorthand_operator)
+            .with_leading_trivia_pieces(token_leading_trivia)
+            .with_trailing_trivia_pieces(token_trailing_trivia);
+
+        let next_node = node
+            .clone()
+            .with_operator_token_token(token)
+            .with_right(state.right.clone());
+
+        mutation.replace_node(
+            AnyJsExpression::JsAssignmentExpression(node.clone()),
+            AnyJsExpression::JsAssignmentExpression(next_node),
+        );
+
+        Some(JsRuleAction {
+            category: ActionCategory::QuickFix,
+            applicability: Applicability::MaybeIncorrect,
+            message: markup! { "Use "<Emphasis>""{shorthand_operator.to_string()?}""</Emphasis>" instead." }
+                .to_owned(),
+            mutation,
+        })
+    }
+}
+
+fn has_same_reference_in_expression(
+    variable_name: String,
```

Comment:
Could we use a string slice instead?
```suggestion
    variable_name: &str,
```